File: domain/src/main/java/net/huray/backend/minuting/minutes/MinutesEntity.java
package net.huray.backend.minuting.minutes;

import static javax.persistence.GenerationType.IDENTITY;

import java.util.ArrayList;
import java.util.List;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import net.huray.backend.minuting.base.BaseDateTimeEntity;

@Getter
@Entity(name = "minutes")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class MinutesEntity extends BaseDateTimeEntity {

    @Id
    @GeneratedValue(strategy = IDENTITY)
    private Long id;

    private String title;

    @OneToMany(mappedBy = "minutes")
    private List<MinutesAttendee> attendees = new ArrayList<>();


}

File: domain/src/main/kotlin/net/huray/backend/minuting/entity/MinutesEntity.kt
package net.huray.backend.minuting.entity

import net.huray.backend.minuting.entity.common.BaseDateTimeEntity
import javax.persistence.*

@Entity(name = "minutes")
class MinutesEntity(
    title: String,
    contents: String,
//    space: SpaceEntity
) : BaseDateTimeEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0L

    var title = title; protected set

    var contents = contents; protected set

    // TODO: 2022/12/21 - 스페이스 스펙 정확히 모르겠어서 일단 lateinit 으로 둠
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "space_id")
    lateinit var space: SpaceEntity// = space; protected set

    @OneToMany(mappedBy = "minutes")
    protected val mutableAttendeeList: MutableList<MinutesAttendee> = mutableListOf()
    val attendeeList: List<MinutesAttendee> get() = mutableAttendeeList.toList()

    fun addAttendee(attendee: MinutesAttendee) {
        mutableAttendeeList.add(attendee)
    }

    @OneToMany(mappedBy = "minutes")
    protected val mutableCommentList: MutableList<CommentEntity> = mutableListOf()
    val commentList: List<CommentEntity> get() = mutableCommentList.toList()

    fun addComment(comment: CommentEntity) {
        mutableCommentList.add(comment)
    }

    fun updateMinutes(title: String, contents: String) {
        this.title = title
        this.contents = contents
    }

}